### PR TITLE
just uploaded from-1D-ascii.ipynb to docs\lib\howto\gamma

### PR DESCRIPTION
**Pull request by guanfada**

### CODE REVIEW

**Comment by SimonBiggs** [comment]
Hi @guanfada,

I'm just making a couple of tweaks, it seems the PR action has got itself into a loop and is repeatedly making commits. I'm fixing that by running the following locally:

```
git pull
git checkout guanfada-patch-1
poetry run pre-commit run --all-files
git add -A
git commit -m "run pre-commit"
git push
```

**Comment by SimonBiggs** [comment]
Hi @guanfada,

This is awesome :) thanks for putting this together. There were a few things I need to fix up just so that it played nice with the docs. For the other maintainers, so they know what I did, here were my steps:

* I cloned the repo locally and ran `pymedphys dev docs`
* This resulted in the following kernel error:

```python
WARNING: /home/simon/git/pymedphys/lib/pymedphys/docs/lib/howto/gamma/from-1D-ascii.ipynb
Traceback (most recent call last):
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_cache/executors/basic.py", line 141, in execute
    yield self.execute_single(
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_cache/executors/basic.py", line 154, in execute_single
    result = single_nb_execution(
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 1093, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/home/simon/.asdf/installs/python/3.9.7/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 543, in async_execute
    async with self.async_setup_kernel(**kwargs):
  File "/home/simon/.asdf/installs/python/3.9.7/lib/python3.9/contextlib.py", line 181, in __aenter__
    return await self.gen.__anext__()
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 504, in async_setup_kernel
    await self.async_start_new_kernel(**kwargs)
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/nbclient/client.py", line 416, in async_start_new_kernel
    await ensure_async(self.km.start_kernel(extra_arguments=self.extra_arguments, **kwargs))
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/nbclient/util.py", line 96, in ensure_async
    result = await obj
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_client/manager.py", line 362, in _async_start_kernel
    raise e
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_client/manager.py", line 351, in _async_start_kernel
    await ensure_async(self._launch_kernel(kernel_cmd, **kw))
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_client/utils.py", line 34, in ensure_async
    return await obj
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_client/manager.py", line 269, in _async_launch_kernel
    connection_info = await self.provisioner.launch_kernel(kernel_cmd, **kw)
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_client/provisioning/local_provisioner.py", line 179, in launch_kernel
    self.process = launch_kernel(cmd, **scrubbed_kwargs)
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_client/launcher.py", line 169, in launch_kernel
    raise ex
  File "/home/simon/.cache/pypoetry/virtualenvs/pymedphys-OJ9w3Q6B-py3.9/lib/python3.9/site-packages/jupyter_client/launcher.py", line 157, in launch_kernel
    proc = Popen(cmd, **kwargs)
  File "/home/simon/.asdf/installs/python/3.9.7/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/simon/.asdf/installs/python/3.9.7/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/simon/git/pymedphys/.venv/bin/python'
```

* As a result, I loaded up the notebook within Jupyter Lab and changed the default kernel to the standard default "Python3" kernel.
* Running the docs then didn't load the files, so I changed some "raw" cells" over to "code cells"
  * Then, everything worked on the notebook end.
* To make the notebook appear in the docs I needed to add an entry to the index,
* Then lastly, I added a .gitignore so as to not commit the files that were downloaded by the notebook.

And, there we have it :slightly_smiling_face:, it worked :slightly_smiling_face:

File: lib/pymedphys/docs/lib/howto/gamma/1D-from-csv.ipynb [hunk 1]
```diff
@@ -0,0 +1,230 @@
+{
```
Comment (SimonBiggs) [inline lib/pymedphys/docs/lib/howto/gamma/1D-from-csv.ipynb:1]: <p>I really liked the improvements to the plot :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymedphys/pymedphys/pull/1603/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (SimonBiggs) [inline lib/pymedphys/docs/lib/howto/gamma/1D-from-csv.ipynb:1]: <p>Yup, awesome.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymedphys/pymedphys/pull/1603/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

**Review by SimonBiggs — APPROVED** [review]
Looks good to me :)

I great addition to the docs :)

Thanks @guanfada

**Comment by guanfada** [comment]
Thank you @SimonBiggs  :-)
One thing I noticed in the global gamma calculation. I found the maximum dose in the reference (measurement) dose set is used for global normalization. But in TG-218 section 2.G.1, it has this statement:
Specifically in global normalization, the dose difference between any measured and calculated dose point pair is
normalized using the same value for all point pairs, often the **maximum planned dose point.**
Note: here planned dose is the "evaluation" dose. 
Would you like to double-check which maximum dose (from reference or evaluation) is used in the gamma() function of PyMedPhys?

**Comment by SimonBiggs** [comment]
> Would you like to double-check which maximum dose (from reference or evaluation) is used in the gamma() function of PyMedPhys?

If no global maximum is provided it appears that yes, it uses the maximum of the reference:

https://github.com/pymedphys/pymedphys/blob/47704682808aa05d218c0f67f16884e3ee76d422/lib/pymedphys/_gamma/implementation/shell.py#L252-L253

Do you think this should be changed to use dose_evaluation as default?

**Comment by SimonBiggs** [comment]
> Do you think this should be changed to use `dose_evaluation` as default?

Would you be interested in making a PR that changes that from `dose_reference` to `dose_evaluation`? I am happy to use the justification for the change to be coming into line with the recommendations with TG-218.

**Comment by guanfada** [comment]
@SimonBiggs 
Sure, I will be happy to make a PR for this change. 
Cheers!

**Comment by SimonBiggs** [comment]
Btw, here's the page live:

https://docs.pymedphys.com/lib/howto/gamma/1D-from-csv.html

**Pull request merged**